packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -0,0 +1,542 @@
+import { once } from 'node:events';
+import { setTimeout } from 'node:timers';
+import { setTimeout as sleep } from 'node:timers/promises';
+import { TextDecoder } from 'node:util';
+import { inflate } from 'node:zlib';
+import { Collection } from '@discordjs/collection';
+import { AsyncQueue } from '@sapphire/async-queue';
+import { AsyncEventEmitter } from '@vladfrangu/async_event_emitter';
+import {
+	GatewayCloseCodes,
+	GatewayDispatchEvents,
+	GatewayIdentifyData,
+	GatewayOpcodes,
+	GatewayReceivePayload,
+	GatewaySendPayload,
+} from 'discord-api-types/v10';
+import { CONNECTING, OPEN, RawData, WebSocket } from 'ws';
+import type { Inflate } from 'zlib-sync';
+import type { SessionInfo } from './WebSocketManager';
+import type { IContextFetchingStrategy } from '../strategies/context/IContextFetchingStrategy';
+import { ImportantGatewayOpcodes } from '../utils/constants';
+import { lazy } from '../utils/utils';
+
+const getZlibSync = lazy(() => import('zlib-sync').then((mod) => mod.default).catch(() => null));
```

Comment:
It's inferior because... it uses ESM like the rest of the codebase (which is also advisable when using typescript, might I add), which our build tool - tsup - will deal with anyways for the cjs builds we provide?